operator/pkg/component/component/component.go

```diff
@@ -82,13 +82,18 @@ type CommonComponentFields struct {
 	// resourceName is the name of all resources for this component.
 	resourceName string
 	// index is the index of the component (only used for components with multiple instances like gateways).
-	index    int
+	index int
+	// spec for the actual gateway, if the component is a gateway.
+	gatewaySpec *v1alpha1.GatewaySpec
```

Comment:
This part doesn't seem to be common for all components? But I understand that this helps to reduce hardcode logic for gateways and components. But just wondering is there better way if possible

Reply:
changed this to interface which i think is a little better. the common go solution in this case is to create an interface and implement it in each struct and store the interface type here but i think this really would be an overkill for this simple situation.